### ISSUE TITLE
`lr_scheduler` has no `state_dict`

### ISSUE THREAD

**Issue by georgeyiasemis** [NKI-AI/direct#217]
**Describe the bug**
'LR scheduler has no state_dict' error appears when training, so it's not checkpointed.

**To Reproduce**
Any training script.

**Expected behavior**
LR scheduler state_dict to be checkpointed and for the error todisappear.
